src/discussions/topics/topic-search-bar/TopicSearchBar.jsx

```diff
@@ -0,0 +1,39 @@
+import { injectIntl, intlShape } from '@edx/frontend-platform/i18n';
+import { SearchField } from '@edx/paragon';
+import { faSort } from '@fortawesome/free-solid-svg-icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import React from 'react';
+import { useDispatch } from 'react-redux';
+import { SelectableDropdown } from '../../../components';
+import { TopicOrdering } from '../../../data/constants';
+import { buildIntlSelectionList } from '../../utils';
+import { setFilter, setSortBy } from '../data';
+import messages from './messages';
+
+function TopicSearchBar({ intl }) {
+  const dispatch = useDispatch();
+  const topicOrderingOptions = buildIntlSelectionList(TopicOrdering, intl, messages);
+  return (
+    <div className="d-flex flex-row p-1 align-items-center">
+      <SearchField
+        className="flex-fill m-1 border-0"
+        placeholder={intl.formatMessage(messages.find_a_topic)}
+        onSubmit={(query) => dispatch(setFilter(query))}
+        onChange={(query) => dispatch(setFilter(query))}
+      />
+      <SelectableDropdown
+        options={topicOrderingOptions}
+        onChange={(option) => dispatch(setSortBy(option.value))}
```

Comment:
seems like it will send an API call on every key change. please use debouncing rule here

Reply:
okay I think its only updating the store and no API call is made so we are good here with this.

Reply:
In either case, this is a dropdown, so can't really debounce, if there is a change, it need to be reflected.